### PR TITLE
i18n(zh-TW): Add translation guide

### PR DESCRIPTION
**Pull request by NightFeather0615**
#### What kind of changes does this PR include?

- Something else!

#### Description

Add translation guide for `zh-tw` i18n team.

The guide is still WIP, this PR should be a draft until every active translator had a chance to review and discuss it.

Todo:
- [ ] Replace example to texts taken from Astro docs.

### CODE REVIEW

File: src/i18n/zh-tw/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+> The mustache tag will be replaced with the value of the msg property from the corresponding component instance. It will also be updated whenever the msg property changes.
+
+可在句中加入如「然而」、「並且」或「即使……也……」等連接詞來使句子更加連貫。  
+
+以上例句的後半句可譯為：「每當 `msg` 屬性發生改變時，綁定處的內容便會隨之更新」。
+
+
+### 3.4 在尊重原文的前提下使讀者更容易理解
+
+> TODO
```
Comment (mingXta) [inline src/i18n/zh-tw/README.md:121]: Did you forget to add it, or do you want to add it later?

File: src/i18n/zh-tw/README.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+|deploy                |部署                 |                       |
+|component             |組件                 |                       |
+|migration             |遷移                 |                       |
+|asset                 |資產                 |                       |
+|docs                  |文檔                 |                       |
+|project               |專案                 |                       |
+|configuration         |配置                 |                       |
+|community             |社群                 |                       |
+|hydration             |水合                 |                       |
+|islands               |群島                 |特指 Astro Islands      |
```
Comment (moneychien19) [inline src/i18n/zh-tw/README.md:37]: `Islands` was not translated to `zh-tw` in the previous doc like [this](https://github.com/withastro/docs/blob/main/src/content/docs/zh-tw/concepts/islands.mdx), so I just wonder if we should translate it in the future?

Comment (NightFeather0615) [reply src/i18n/zh-tw/README.md:37]: Yes, I have a few considerations:
- Most i18n teams have translated `islands` into their own language
- The word "群島" matches the original meaning well enough to not cause confusion
- The `zh-cn` team also translates `islands`, I want to minimize the confusion for Chinese speakers

Comment (moneychien19) [reply src/i18n/zh-tw/README.md:37]: Thanks for replying! Everything LGTM in this PR.

**Comment by sarah11918** [comment]
Hi there! I love that you have decided to make your own translation guide! 

Instead of a draft PR, merging something smaller to start, that can be PRd and improved upon might be more helpful. I would be happy to approve merging something and then you can continue to add/improve it! Please let a maintainer know when you feel ready to do this by undrafting the PR , and we will merge it!  :raised_hands:

**Comment by NightFeather0615** [comment]
Sure! I've undraft this PR, it should be ready for merging, and I hope to let other `zh-tw` contributors maintain this guide, as I've been busy with some school stuff recently 😵.

**Comment by sarah11918** [comment]
So happy to get this merged! Thank you @NightFeather0615 !

**Pull request merged**